pkg/job_controller/job_controller.go

```diff
@@ -187,3 +190,114 @@ func (jc *JobController) resolveControllerRef(namespace string, controllerRef *m
 	}
 	return job
 }
+
+func (jc *JobController) createCache(metaObject metav1.Object, cacheBackendSpec *cachev1alpha1.CacheBackendSpec) error {
+	cacheBackend := &cachev1alpha1.CacheBackend{}
+	cacheBackendName := cachectrl.GetCacheName(metaObject)
+	cacheBackendNameSpace := metaObject.GetNamespace()
+	err := jc.Client.Get(context.Background(), types.NamespacedName{
+		Namespace: cacheBackendNameSpace,
+		Name:      cacheBackendName,
+	}, cacheBackend)
+
+	if err == nil {
+		// Already been created
+		log.Infof("cache backend has been created")
+		return nil
+	} else {
+		if k8serrors.IsNotFound(err) {
+			log.Infof("cache backend is not exist, start to create %s", cacheBackendName)
+
+			// If haven't created yet
+			cacheBackend = &cachev1alpha1.CacheBackend{}
+			cacheBackend.Name = cacheBackendName
+			cacheBackend.Namespace = cacheBackendNameSpace
+			cacheBackend.Spec = *cacheBackendSpec
+			controllerRef := jc.GenOwnerReference(metaObject)
```

Comment:
Yes, we can leave the ownerReference out so that CacheBackend can be retained for reuse when jobs are deleted.

But now that some of the parameters of cacheBackend are bound to the job, such as the cachebackend name. we can achieve reuse by having the user specify the name of cacheBackend and add some field.

Reply:
yep, so the YAML spec needs to support specifying the name, and if name is speficied, we will not delete the cache when the job is done. Or we could have a explicit flag 'retained'. I think the latter is better. 
And the default behavior could be retained, if a name is speficied